modules/javafx.base/src/main/java/javafx/beans/value/ChangeListener.java

```diff
@@ -48,8 +48,13 @@
     /**
      * Called when the value of an {@link ObservableValue} changes.
      * <p>
-     * In general, it is considered bad practice to modify the observed value in
-     * this method.
+     * Changing the observed value in this method will result in all listeners
+     * being notified of this latest change after the initial change
+     * notification (with the original old and values) has completed.
+     * The listeners that still needed to be notified may see a new value that
+     * differs from a call to {@link ObservableValue#getValue}. All listeners are
+     * then notified again with an old value equal to the initial new value,
+     * and a new value with the latest value.
```

Comment:
This seems like a right place.
It is still a bad practice to change the value in this method, so I would recommend to keep the warning.
I did little rewording, please check how does this look:

```
* In general, it is considered a bad practice to modify the observed value in
* this method. But if still modified, then the notifications for this change
* are deferred until all the listeners are notified of current change.
* This results in a minor inconsistency of new value with the listeners
* that are still pending to be notified of current change that the listeners
* would receive a new value that differs from a call to {@link ObservableValue#getValue}.
```